django/db/models/functions/comparison.py

```diff
@@ -46,7 +46,7 @@ def as_postgresql(self, compiler, connection, **extra_context):
         # CAST would be valid too, but the :: shortcut syntax is more readable.
         # 'expressions' is wrapped in parentheses in case it's a complex
         # expression.
-        return self.as_sql(compiler, connection, template='(%(expressions)s)::%(db_type)s', **extra_context)
+        return self.as_sql(compiler, connection, template='((%(expressions)s)::%(db_type)s)', **extra_context)
```

Comment:
Since we used the `::%(db_type)s` for PostgreSQL solely for readability purposes and this now requires two parentheses wrapping `((expr)::type)` I think we should consider removing that `as_postgresql` override entirely as `CAST(expr as type)` seems more readable to me.

Reply:
I agree with Simon.

Reply:
This would solve the original problem as well. Two tests fail without the `as_postgresql` override, but there are looking for `::type` in the generated queries.